pennylane/transforms/qcut.py

```diff
@@ -939,6 +942,367 @@ def qcut_processing_fn_mc(
+        with a qnode, this shots value is automatically inferred from the provided
+        device.
+
+        Next, each circuit fragment is randomly expanded over :class:`~.MeasureNode` and
+        :class:`~.PrepareNode` configurations. For each pair, a measurement is sampled from
+        the Pauli basis and a state preparation is sampled from the corresponding pair of eigenstates.
+
+        A settings array is also given which tracks the configuration pairs. Since each of the 4
+        measurements has 2 possible eigenvectors, all configurations can be uniquely identified by
+        8 values. The number of rows is determined by the number of cuts and the number of columns
+        is determined by the number of shots.
+
+        >>> shots = 3
+        >>> configurations, settings = qml.transforms.qcut.expand_fragment_tapes_mc(
+        ...        fragment_tapes, communication_graph, shots=shots
+        ...    )
+        >>> tapes = tuple(tape for c in configurations for tape in c)
+        >>> settings
+        tensor([[0, 4, 7]], requires_grad=True)
+
+        Each configuration is drawn below:
+
+        >>> for t in tapes:
+        ...     print(t.draw())
+        ...     print("")
+
+        .. code-block::
+
+            0: ──H─╭C────┤  Sample[|1⟩⟨1|]
+            1: ────╰X──X─┤  Sample[I]
+
+            0: ──H─╭C────┤  Sample[|1⟩⟨1|]
+            1: ────╰X──X─┤  Sample[Y]
+
+            0: ──H─╭C────┤  Sample[|1⟩⟨1|]
+            1: ────╰X──X─┤  Sample[Z]
+
+            0: ──I─╭C─┤  Sample[|1⟩⟨1|]
+            1: ────╰X─┤  Sample[|1⟩⟨1|]
+
+            0: ──H──S─╭C─┤  Sample[|1⟩⟨1|]
+            1: ───────╰X─┤  Sample[|1⟩⟨1|]
+
+            0: ──X─╭C─┤  Sample[|1⟩⟨1|]
+            1: ────╰X─┤  Sample[|1⟩⟨1|]
+
+        The last step is to execute the tapes and postprocess the results using
+        :func:`~.qcut_processing_fn_sample`, which processes the results to approximate the original full circuit
+        output bitstrings.
+
+        >>> results = qml.execute(tapes, dev, gradient_fn=None)
+        >>> qml.transforms.qcut.qcut_processing_fn_sample(
+        ...     results,
+        ...     communication_graph,
+        ...     shots=shots,
+        ... )
+        [array([[1., 0., 0.],
+                [0., 0., 0.],
+                [1., 1., 1.]])]
+
+        Alternatively, it is possible to calculate an expectation value if a classical
+        processing function is provided that will accept the reconstructed circuit bitstrings
+        and return a value in the interval :math:`[-1, 1]`:
+
+        .. code-block::
+
+            def fn(x):
+                if x[0] == 0:
+                    return 1
+                if x[0] == 1:
+                    return -1
+
+        >>> qml.transforms.qcut.qcut_processing_fn_mc(
+        ...     results,
+        ...     communication_graph,
+        ...     settings,
+        ...     shots,
+        ...     fn
+        ... )
+        array(4.)
```

Comment:
It looks weird that the estimate is outside `[-1, 1]`, but I think the estimator just has a large variance :thinking:

Reply:
The shots is also very small (`3`)